apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -248,7 +249,10 @@ def _convert_includes(self, project_dir, source_dir):
         for ext in ("c", "h", "cpp"):
             for filename in source_dir.rglob(f"*.{ext}"):
                 with filename.open() as file:
-                    lines = file.readlines()
+                    try:
+                        lines = file.readlines()
+                    except:
```

Comment:
@mehrdadh OK. So what I suggest is to catch that exception specifically (`except UnicodeDecodeError:`) and add a comment like a TODO() explaning that we don't get this error on Arduino tests and that needs further investigation. Also since iirc the linter will consider it (not catching a specific exception) an anti-pattern.

Reply:
done.